### ISSUE TITLE
Create documentation describing Linux kernel security options

### ISSUE THREAD

**Issue by a13xp0p0v** [a13xp0p0v/kconfig-hardened-check#69]

**Comment by o8opi** [comment]
Would love to see this, even if it's just a list of links and pointers to other resources :)

**Comment by jbmaillet** [comment]
https://kernsec.org/wiki/index.php/Kernel_Self_Protection_Project/Recommended_Settings

Especially this page, but actually the whole site, an initiative from Kernel security maintainer Kees Cook:
https://lore.kernel.org/kernel-hardening/CAGXu5jJ3FgxXK9WuOLRwnEq=y4dS+CTm+WQBxWe3sYZ7e9p6Gg@mail.gmail.com/

**Comment by a13xp0p0v** [comment]
@krishjainx, @jbmaillet, yes, I mean creating the documentation describing how the checked parameters influence Linux kernel security.

Another good example is CLIP OS documentation: https://docs.clip-os.org/clipos/kernel.html#configuration

I think of creating `doc` directory with markdown files describing Kconfig options, kernel cmdline arguments, and sysctl parameters.

**Comment by a13xp0p0v** [comment]
@krishjainx , yes, some part of this work can be automated.

For `self_protection`, `security_policy`, and `harden_userspace` parameters, the Kconfig descriptions and [kernel documentation](https://docs.kernel.org/admin-guide) contain some security-relevant info.
Example: https://cateee.net/lkddb/web-lkddb/CFI_CLANG.html

But for `cut_attack_surface` parameters, the kernel documentation doesn't say much about the security implications.

**Comment by jvoisin** [comment]
> But for cut_attack_surface parameters, the kernel documentation doesn't say much about the security implications.

I think it would make sense to add some info upstream in the Kconfig description. Ideally we should be able to run a glorified `grep` on the Kconfig and generate proper documentation.

**Comment by jo-so** [comment]
There is also [Linux Hardening Guide | Madaidan's Insecurities](https://madaidans-insecurities.github.io/guides/linux-hardening.html) which describes options like [hidepid](https://madaidans-insecurities.github.io/guides/linux-hardening.html#hidepid).
